### ISSUE TITLE
Issues getting prettier to load the plugin

### ISSUE THREAD

**Issue by feross** [brodybits/prettier-plugin-prettierx-babel#3]
Hi there! I installed `prettier` and `prettier-plugin-x` as dev dependencies. When I run `npx prettier --help`, I get output with the prettierx rules, which indicates that prettier has found the plugin.

But then when I run: `npx prettier --write path/to/my/file.js`, it appears that the plugin rules are ignored while the built-in prettier rules are applied without issues.

Any idea of what could be going on here?

**Comment by feross** [comment]
Here's the output of `npm ls` to show that they're installed next to each other in node_modules:

```
npm ls prettier
sync@1.0.0 /Users/feross/code/sync
├─┬ prettier-plugin-x@0.0.6
│ └── prettier@2.2.1 deduped
└── prettier@2.2.1

npm ls prettier-plugin-x
sync@1.0.0 /Users/feross/code/sync
└── prettier-plugin-x@0.0.6
```

**Comment by brody2consult** [comment]
The trick is that you have to configure Prettier to use the right parser, which would be `x-babel` for JavaScript.

I hacked together `prettier-config-x` which configures the overrides to use `x-babel` for JavaScript files and `x-babel-ts` for TypeScript files, hope I got it right.

**Comment by feross** [comment]
Is the happy path to just use `prettier-config-x-standard`?

Also, what is `x-babel`? I can't seem to find a reference to it anywhere.

**Comment by brody2consult** [comment]
> Is the happy path to just use `prettier-config-x-standard`?

Yes, hopefully. Unfortunately there is divergence between `standard` and `prettier-config-x-standard` in the formatting of ternaries due to issues on both sides:

- https://github.com/standard/standard/issues/1624 which is due to issues I referenced on eslint
- https://github.com/brodybits/prettierx/issues/468

> Also, what is `x-babel`?

`x-babel` would be the custom parser that exports the `babel` parser from the `prettierx` package. It is referenced in the Sample Usage section but not very clearly: https://github.com/brodybits/prettierx/issues/468

It is here in `index.js`: https://github.com/brodybits/prettier-plugin-x/blob/0.0.6/index.js#L11

Documentation definitely needs improvement.

**Comment by feross** [comment]
Oh, interesting. I'm not really familiar with the inner workings of `prettier` or `prettierx`. Why is a custom parser needed?

Anyway, we were able to get `prettier-confix-x-standard` working – hooray! It would be great to figure out how to get prettier support integrated directly into `standard`. This is something I'd love to try to get into the next version of `standard`. Perhaps we could collaborate on it when we both have time?

**Comment by brody2consult** [comment]
> Why is a custom parser needed?

Here is my understanding:

By design Prettier expects the plugin to specify which parser is connected to which printer. It does not expect a plugin to modify the default parser or printer for existing languages such as JavaScript.

This is where the prettier-config packages can come handy ... assuming we get it right. Unfortunately the only way they support *extending* shared config extensions is with an object extension hack like this: https://github.com/brodybits/prettier-config-x-standard/blob/0.0.2/index.js#L2

> ```js
> module.exports = {
>   ...require("prettier-config-x"),
>   arrowParens: "avoid",
>   semi: false,
>   singleQuote: true,
>   jsxSingleQuote: true,
>   trailingComma: "none",
>   generatorStarSpacing: true,
>   offsetTernaryExpressions: true,
>   spaceBeforeFunctionParen: true,
>   yieldStarSpacing: true,
> };
> ```

> we were able to get `prettier-confix-x-standard` working – hooray!

If you did this in a PR, I would love to check it out.

> This is something I'd love to try to get into the next version of `standard`. Perhaps we could collaborate on it when we both have time?

I would definitely love to work with you on this and can open an issue on `standard` for discussion, if you like.

**Comment by makcim392** [comment]
> > Why is a custom parser needed?
> 
> Here is my understanding:
> 
> By design Prettier expects the plugin to specify which parser is connected to which printer. It does not expect a plugin to modify the default parser or printer for existing languages such as JavaScript.
> 
> This is where the prettier-config packages can come handy ... assuming we get it right. Unfortunately the only way they support _extending_ shared config extensions is with an object extension hack like this: https://github.com/brodybits/prettier-config-x-standard/blob/0.0.2/index.js#L2
> 
> > ```js
> > module.exports = {
> >   ...require("prettier-config-x"),
> >   arrowParens: "avoid",
> >   semi: false,
> >   singleQuote: true,
> >   jsxSingleQuote: true,
> >   trailingComma: "none",
> >   generatorStarSpacing: true,
> >   offsetTernaryExpressions: true,
> >   spaceBeforeFunctionParen: true,
> >   yieldStarSpacing: true,
> > };
> > ```
> 
> > we were able to get `prettier-confix-x-standard` working – hooray!
> 
> If you did this in a PR, I would love to check it out.
> 
> > This is something I'd love to try to get into the next version of `standard`. Perhaps we could collaborate on it when we both have time?
> 
> I would definitely love to work with you on this and can open an issue on `standard` for discussion, if you like.

Hi, I'm using Angular with TS, where does "**module.exports = {**" go?
